pytensor/tensor/extra_ops.py

```diff
@@ -1554,9 +1560,26 @@ def broadcast_shape_iter(
                 result_dims.append(first_length)
                 continue
 
-            # Add assert that all remaining shapes are equal
-            condition = pt_all([pt_eq(first_length, other) for other in other_lengths])
-            result_dims.append(_broadcast_assert(first_length, condition))
+            if not allow_runtime_broadcast:
+                # Add assert that all remaining shapes are equal
+                condition = pt_all(
+                    [pt_eq(first_length, other) for other in other_lengths]
+                )
+                result_dims.append(_broadcast_assert(first_length, condition))
+            else:
+                lengths = as_tensor_variable((first_length, *other_lengths))
+                runtime_broadcastable = pt_eq(lengths, one)
+                result_dim = pt_abs(
+                    pt_max(switch(runtime_broadcastable, -one, lengths))
```

Comment:
Fun trick with the -one. I guess this assumes that we always want to store shapes as signed integers. I think that's fine though.

Reply:
Shouldn't matter, the one is an int64 constant, so that's not going anywhere. Or did you mean this would upcast if the shapes were to be uint64 (by default they are always int64)?